radixdlt/src/main/java/com/radixdlt/consensus/sync/SyncedRadixEngine.java

```diff
@@ -162,15 +165,19 @@ public boolean syncTo(VertexMetadata vertexMetadata, List<ECPublicKey> target, O
 			return true;
 		}
 
-		// TODO: better randomization of peer selection
-		Peer peer = target.stream()
-			.map(pk -> addressBook.peer(pk.euid()))
-			.filter(Optional::isPresent)
-			.map(Optional::get)
-			.filter(Peer::hasSystem)
-			.findFirst()
-			.orElseThrow(() -> new RuntimeException("Unable to find peer"));
-		stateSyncNetwork.sendSyncRequest(peer, currentStateVersion);
+		syncManager.syncToVersion(targetStateVersion, version -> {
+			List<Peer> peers = target.stream()
+					.map(pk -> addressBook.peer(pk.euid()))
+					.filter(Optional::isPresent)
+					.map(Optional::get)
+					.filter(Peer::hasSystem)
+					.collect(Collectors.toList());
+			Peer peer = peers.stream()
```

Comment:
The use of streams to select a random item from a list seems wasteful and makes it harder to understand than the straightforward approach.

Please use:
```
Peer peer = peers.get(ThreadLocalRandom.current().nextInt(peers.size()));
```

Reply:
Ok.